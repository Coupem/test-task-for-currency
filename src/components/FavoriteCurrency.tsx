// eslint-disable-next-line @typescript-eslint/no-use-before-define
import React from 'react';
import PropTypes from 'prop-types';

import { ListGroup } from 'react-bootstrap';
import {
  FavoriteCurrencyWrapper,
  NoFavoriteCurrency,
  TitleFavoriteCurrency,
  ListGroupContainer,
} from '../styled';
import { FavoriteCurrencyProp } from '../types';

const FavoriteCurrency = ({
  favoriteCurrencies,
}: FavoriteCurrencyProp): JSX.Element => (
  <FavoriteCurrencyWrapper>
    <TitleFavoriteCurrency>Favorite Currency:</TitleFavoriteCurrency>

    {Object.values(favoriteCurrencies).length === 0 && (
      <NoFavoriteCurrency>NO CURRENCY</NoFavoriteCurrency>
    )}

    {Object.values(favoriteCurrencies).map((breakpoint) => (
      <ListGroupContainer key={breakpoint.ccy}>
        <ListGroup.Item>Currency: {breakpoint.ccy}</ListGroup.Item>
        <ListGroup.Item>Buy: {breakpoint.buy}</ListGroup.Item>
        <ListGroup.Item>Sale: {breakpoint.sale}</ListGroup.Item>
      </ListGroupContainer>
    ))}
  </FavoriteCurrencyWrapper>
);

FavoriteCurrency.propTypes = {
  favoriteCurrencies: PropTypes.shape({
    ccy: PropTypes.string,
    buy: PropTypes.string,
    sale: PropTypes.string,
  }),
};

export default FavoriteCurrency;
