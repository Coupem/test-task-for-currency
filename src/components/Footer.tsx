import React from 'react';
import { FooterComponent } from '../styled';

const Footer = (): JSX.Element => (
  <FooterComponent>Â© 2021 My. Built My Theme.</FooterComponent>
);

Footer.propTypes = {};

export default Footer;
