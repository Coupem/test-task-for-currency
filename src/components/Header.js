import React from 'react';
import { HeaderComponent } from '../styled';

const Header = () => (
  <HeaderComponent>Best converter & valid currency course</HeaderComponent>
);

Header.propTypes = {};

export default Header;
