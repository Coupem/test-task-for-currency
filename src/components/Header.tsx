// eslint-disable-next-line @typescript-eslint/no-use-before-define
import React from 'react';
import { HeaderComponent } from '../styled';

const Header = (): JSX.Element => (
  <HeaderComponent>Best converter & valid currency course</HeaderComponent>
);

Header.propTypes = {};

export default Header;
