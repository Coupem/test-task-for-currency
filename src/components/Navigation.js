import React from 'react';
import { useHistory, useLocation } from 'react-router-dom';

import { CONVERTER_URL, COURSE_URL } from '../constants';
import { NavButton, NavigationComponent } from '../styled';

const Navigation = () => {
  const history = useHistory();
  const { pathname } = useLocation();

  const onClickButtonConverter = () => {
    history.push('/');
  };

  const onClickButtonCourse = () => {
    history.push('/course');
  };

  return (
    <NavigationComponent>
      <NavButton
        primary={pathname === CONVERTER_URL}
        onClick={onClickButtonConverter}
      >
        Converter
      </NavButton>
      <NavButton
        primary={pathname === COURSE_URL}
        onClick={onClickButtonCourse}
      >
        Course
      </NavButton>
    </NavigationComponent>
  );
};

Navigation.propTypes = {};

export default Navigation;
