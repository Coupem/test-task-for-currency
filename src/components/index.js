export { default as InputValuesGroup } from './Converter/InputValuesGroup';
export { default as ConvertCurrency } from './Converter/ConvertCurrency';
export { default as FavoriteCurrency } from './Course/FavoriteCurrency';
export { default as Navigation } from './Navigation/Navigation';
export { default as Header } from './Header/Header';
export { default as Footer } from './Footer/Footer';
