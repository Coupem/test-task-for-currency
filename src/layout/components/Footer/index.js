import React from 'react';
import { FooterComponent } from './styledComponents';

function Footer() {
  return <FooterComponent>Â© 2021 My. Built My Theme.</FooterComponent>;
}

Footer.propTypes = {};

export default Footer;
