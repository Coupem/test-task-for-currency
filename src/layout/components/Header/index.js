import React from 'react';
import { HeaderComponent } from './styledComponents';

function Header() {
  return (
    <HeaderComponent>Best converter & valid currency course</HeaderComponent>
  );
}

Header.propTypes = {};

export default Header;
