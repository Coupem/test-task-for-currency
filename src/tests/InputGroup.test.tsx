import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import { InputValuesGroup } from '../components/index';

// describe('Footer', () => {
//   let wrapper;
//   beforeAll(() => {
//     const wrapper = shallow(<InputValuesGroup {...props} />);
//   });
//
//   it('should render correctly', () => {
//     expect(toJson(wrapper)).toMatchSnapshot();
//   });
// });
