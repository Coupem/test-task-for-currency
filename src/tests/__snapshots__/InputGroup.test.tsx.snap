// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <InputValuesGroup
    baseCurrency="string"
    currency={
      Object {
        "currency": Array [
          Object {
            "baseCcy": "UAH",
            "buy": "27.05000",
            "ccy": "USD",
            "sale": "27.45000",
          },
          Object {
            "baseCcy": "UAH",
            "buy": "31.85000",
            "ccy": "EUR",
            "sale": "32.45000",
          },
          Object {
            "baseCcy": "UAH",
            "buy": "1",
            "ccy": "UAH",
            "sale": "1",
          },
          Object {
            "baseCcy": "UAH",
            "buy": "0.35000",
            "ccy": "RUR",
            "sale": "0.38000",
          },
          Object {
            "baseCcy": "USD",
            "buy": "31125.3298",
            "ccy": "BTC",
            "sale": "34401.6803",
          },
        ],
        "error": "string",
        "isLoadingCurrency": false,
      }
    }
    handleChangeCurrency={[Function]}
    handleChangeInput={[Function]}
  >
    <Styled(Component)
      className="m-auto"
    >
      <ForwardRef(InputGroup)
        className="sc-fnVZcZ ceSpry m-auto"
      >
        <div
          className="sc-fnVZcZ ceSpry m-auto input-group"
        >
          <Formik
            initialValues={
              Object {
                "amount": "",
                "currency": "string",
              }
            }
            onSubmit={[Function]}
            validate={[Function]}
          >
            <form>
              <styled.input
                data-testid="input"
                name="amount"
                onChange={[Function]}
                placeholder="input value"
                type="number"
                value=""
              >
                <input
                  className="sc-hBMUJo gCIqZs"
                  data-testid="input"
                  name="amount"
                  onChange={[Function]}
                  placeholder="input value"
                  type="number"
                  value=""
                />
              </styled.input>
              <styled.select
                defaultValue="string"
                name="currency"
                onChange={[Function]}
              >
                <select
                  className="sc-bkbkJK bTnLfD"
                  defaultValue="string"
                  name="currency"
                  onChange={[Function]}
                >
                  <styled.option
                    key="USD"
                    value="USD"
                  >
                    <option
                      className="sc-fFSPTT kWHaHm"
                      value="USD"
                    >
                      USD
                    </option>
                  </styled.option>
                  <styled.option
                    key="EUR"
                    value="EUR"
                  >
                    <option
                      className="sc-fFSPTT kWHaHm"
                      value="EUR"
                    >
                      EUR
                    </option>
                  </styled.option>
                  <styled.option
                    key="UAH"
                    value="UAH"
                  >
                    <option
                      className="sc-fFSPTT kWHaHm"
                      value="UAH"
                    >
                      UAH
                    </option>
                  </styled.option>
                  <styled.option
                    key="RUR"
                    value="RUR"
                  >
                    <option
                      className="sc-fFSPTT kWHaHm"
                      value="RUR"
                    >
                      RUR
                    </option>
                  </styled.option>
                  <styled.option
                    key="BTC"
                    value="BTC"
                  >
                    <option
                      className="sc-fFSPTT kWHaHm"
                      value="BTC"
                    >
                      BTC
                    </option>
                  </styled.option>
                </select>
              </styled.select>
            </form>
          </Formik>
        </div>
      </ForwardRef(InputGroup)>
    </Styled(Component)>
  </InputValuesGroup>
</Provider>
`;
